httpserver/src/main/kotlin/tbdex/sdk/httpserver/handlers/SubmitClose.kt

```diff
@@ -0,0 +1,73 @@
+package tbdex.sdk.httpserver.handlers
+
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.request.*
+import io.ktor.server.response.*
+import tbdex.sdk.httpclient.models.ErrorDetail
+import tbdex.sdk.httpserver.models.CallbackError
+import tbdex.sdk.httpserver.models.ErrorResponse
+import tbdex.sdk.httpserver.models.ExchangesApi
+import tbdex.sdk.httpserver.models.SubmitCallback
+import tbdex.sdk.protocol.models.Close
+import tbdex.sdk.protocol.models.Message
+import tbdex.sdk.protocol.models.MessageKind
+
+/**
+ * Handles the submission of a close message by parsing the incoming message,
+ * validating the submission, and executing the specified callback if provided.
+ *
+ * @param call The [ApplicationCall] instance representing the HTTP call.
+ * @param exchangesApi The [ExchangesApi] instance for interacting with TBDex exchanges.
+ * @param callback The optional callback function to be executed after a successful close submission.
+ */
+@Suppress("TooGenericExceptionCaught", "MaxLineLength")
+suspend fun submitClose(
+  call: ApplicationCall,
+  exchangesApi: ExchangesApi,
+  callback: SubmitCallback?
+) {
+  val message: Close?
+
+  try {
+    message = Message.parse(call.receiveText()) as Close
+  } catch (e: Exception) {
+    val errorDetail = ErrorDetail(detail = "Parsing of TBDex message failed: ${e.message}")
+    val errorResponse = ErrorResponse(listOf(errorDetail))
+    call.respond(HttpStatusCode.BadRequest, errorResponse)
+    return
+  }
+
+  val exchangeId = message.metadata.exchangeId.toString()
+  val exchange = exchangesApi.getExchange(exchangeId)
+  if (exchange == null) {
+    val errorDetail = ErrorDetail(detail = "Could not find exchange: $exchangeId")
+    call.respond(HttpStatusCode.NotFound, ErrorResponse(listOf(errorDetail)))
+    return
+  }
+
+  if (!exchange.last().validNext.contains(MessageKind.close)) {
```

Comment:
`last()` can throw a `NoSuchElementException` when the list is empty. Perhaps this code should call `lastOrNull()`?

Reply:
will check for null or empty above